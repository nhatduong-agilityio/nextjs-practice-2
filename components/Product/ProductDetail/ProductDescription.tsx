import { Flex, Heading, ListItem, Stack, Text, UnorderedList, VStack } from '@chakra-ui/react'
import React, { memo } from 'react'

type TDetail = {
  id: string
  content: string
}

interface IProps {
  description: string
  listDetail: TDetail[]
}

const ProductDescription = memo(({ description, listDetail }: IProps) => (
  <Flex w='full' flexDir={['column', 'row']} pt={[0, '30px']}>
    <VStack w={['full', '50%']} style={{ margin: 0 }}>
      <Heading
        w='full'
        variant='large'
        fontSize='30px'
        display={['none', 'flex']}
        pb='25px'
        lineHeight='inherit'
      >
        Item Detail
      </Heading>
      <Text w='full' size={['16', '20']} variant='grayScheme'>
        {description}
      </Text>
    </VStack>
    <Stack w={['full', '50%']} pl={[0, '50px']} pt={[0, '60px']}>
      <UnorderedList gap='10px' mt={['20px', 0]}>
        {listDetail.map((detail) => (
          <ListItem key={detail.id} pb={['15px', '25px']}>
            <Text size={['16', '20']} variant='grayScheme'>
              {detail.content}
            </Text>
          </ListItem>
        ))}
      </UnorderedList>
    </Stack>
  </Flex>
))

ProductDescription.displayName = 'ProductProductDescription'

export default ProductDescription
