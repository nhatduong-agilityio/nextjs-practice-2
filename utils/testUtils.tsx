import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
