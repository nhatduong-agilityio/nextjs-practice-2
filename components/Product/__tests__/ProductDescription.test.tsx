// Utils
import { render } from '@utils/testUtils'

// Components
import ProductDescription from '../ProductDetail/ProductDescription'

describe('ProductDescription render', () => {
  const propsMock = {
    description: 'Product description mock',
    listDetail: [
      { id: '1', content: 'High Quality Leathers4' },
      { id: '2', content: 'High Quality Paints4' },
    ],
  }

  it('Should ProductDescription match Snapshot', () => {
    const { container } = render(<ProductDescription {...propsMock} />)

    expect(container).toMatchSnapshot()
  })
})
