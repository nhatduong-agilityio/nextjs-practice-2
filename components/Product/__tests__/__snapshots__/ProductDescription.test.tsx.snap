// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDescription render Should ProductDescription match Snapshot 1`] = `
<div>
  <div
    class="css-kx2jl1"
  >
    <div
      class="chakra-stack css-2ogt3p"
      style="margin: 0px;"
    >
      <h2
        class="chakra-heading css-m47qtn"
      >
        Item Detail
      </h2>
      <p
        class="chakra-text css-asz83u"
      >
        Product description mock
      </p>
    </div>
    <div
      class="chakra-stack css-lw9l6d"
    >
      <ul
        class="css-1887c3j"
        role="list"
      >
        <li
          class="css-1cr0hgh"
        >
          <p
            class="chakra-text css-1l9mnve"
          >
            High Quality Leathers4
          </p>
        </li>
        <li
          class="css-1cr0hgh"
        >
          <p
            class="chakra-text css-1l9mnve"
          >
            High Quality Paints4
          </p>
        </li>
      </ul>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
